src/Framework/FieldsAPI/Concerns/SerializeAsJson.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Give\Framework\FieldsAPI\Concerns;
+
+use JsonSerializable;
+
+trait SerializeAsJson {
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function jsonSerialize() {
+		return array_merge(
+			// These values must be serialized for all types
+			[
+				'type' => $this->getType(),
+				'name' => $this->getName(),
+			],
+			// We (recursively) serialize all of the class’ properties and exclude the list provided.
+			array_map(
+				static function ( $value ) {
+					if ( $value instanceof JsonSerializable ) {
+						return $value->jsonSerialize();
+					}
+					return $value;
+				},
+				// Only serialize properties which are not hidden.
+				array_diff_key(
+					get_object_vars( $this ),
+					// Respect hidden, if set.
+					array_flip( property_exists( $this, 'hidden' ) ? $this->hidden : [] ),
+					// Ignore hidden property.
+					[ 'hidden' => 'hidden' ]
```

Comment:
I don' think this is being used at all, but it was a way to hide attributes from serialization since right now every attribute on the class is being serialized. The inspiration was from Eloquent models and their `$hidden` array attribute.

Reply:
I'm going to remove this for now, since it isn't being used.

Reply:
Removed in 81ca7e5

Reply:
Yeah, I like the idea. Let's add it when we need it.